Google.Api.Generator/ProtoUtils/ProtoCatalog.cs

```diff
@@ -38,18 +38,22 @@ public ProtoCatalog(string defaultPackage, IEnumerable<FileDescriptor> descs)
                     (field, res: ResourceDetails.LoadResourceReference(msg, field, resourcesByUrt)))
                     .Where(x => x.res != null))
                 .ToDictionary(x => x.field.FullName, x => x.res);
+            _commonUrts = resourcesByUrt.Values.Where(x => x.IsCommon).Select(x => x.UnifiedResourceTypeName).ToImmutableHashSet();
         }
 
         private readonly string _defaultPackage;
         private readonly IReadOnlyDictionary<string, MessageDescriptor> _msgs;
         private readonly IReadOnlyDictionary<string, ResourceDetails.Field> _resourcesByFieldName;
         private readonly IReadOnlyDictionary<string, IReadOnlyList<ResourceDetails.Definition>> _resourcesByFileName;
+        private readonly IImmutableSet<string> _commonUrts;
 
         public MessageDescriptor GetMessageByName(string name) => _msgs[name.Contains('.') ? name : $"{_defaultPackage}.{name}"];
 
         public ResourceDetails.Field GetResourceDetailsByField(FieldDescriptor fieldDesc) => _resourcesByFieldName.GetValueOrDefault(fieldDesc.FullName);
 
         public IEnumerable<ResourceDetails.Definition> GetResourceDefsByFile(FileDescriptor fileDesc) =>
             _resourcesByFileName.GetValueOrDefault(fileDesc.Name, ImmutableList<ResourceDetails.Definition>.Empty);
+
+        public bool IsCommonUrt(string type) => _commonUrts.Contains(type);
```

Comment:
Perhaps IsCommonResourceType? "Urt" isn't terribly obvious unless you already know the context.

Reply:
Done